hazelcast/src/main/java/com/hazelcast/instance/impl/BootstrappedJetProxy.java

```diff
@@ -150,68 +149,69 @@ public <K, V> ReplicatedMap<K, V> getReplicatedMap(@Nonnull String name) {
     @Nonnull
     @Override
     public JetCacheManager getCacheManager() {
-        return jet.getCacheManager();
+        return jetInstance.getCacheManager();
     }
 
     @Nonnull
     @Override
     public <E> IList<E> getList(@Nonnull String name) {
-        return jet.getList(name);
+        return jetInstance.getList(name);
     }
 
     @Nonnull
     @Override
     public <T> ITopic<T> getReliableTopic(@Nonnull String name) {
-        return jet.getReliableTopic(name);
+        return jetInstance.getReliableTopic(name);
     }
 
     @Nonnull
     @Override
     public <T> Observable<T> getObservable(@Nonnull String name) {
-        return jet.getObservable(name);
+        return jetInstance.getObservable(name);
     }
 
     @Override
     public void shutdown() {
-        jet.shutdown();
+        jetInstance.shutdown();
     }
 
     @Override
     public boolean existsDistributedObject(@Nonnull String serviceName, @Nonnull String objectName) {
-        return jet.existsDistributedObject(serviceName, objectName);
+        return jetInstance.existsDistributedObject(serviceName, objectName);
     }
 
     @Override
     public ILogger getLogger() {
-        return jet.getLogger();
+        return jetInstance.getLogger();
     }
 
     @Override
     public Job newJobProxy(long jobId, M lightJobCoordinator) {
-        return jet.newJobProxy(jobId, lightJobCoordinator);
+        return jetInstance.newJobProxy(jobId, lightJobCoordinator);
     }
 
     @Override
     public Job newJobProxy(long jobId, boolean isLightJob, @Nonnull Object jobDefinition, @Nonnull JobConfig config) {
-        return jet.newJobProxy(jobId, isLightJob, jobDefinition, config);
+        return jetInstance.newJobProxy(jobId, isLightJob, jobDefinition, config);
     }
 
     @Override
     public M getMasterId() {
-        return jet.getMasterId();
+        return jetInstance.getMasterId();
     }
 
     @Override
     public Map<M, GetJobIdsOperation.GetJobIdsResult> getJobsInt(String onlyName, Long onlyJobId) {
-        return jet.getJobsInt(onlyName, onlyJobId);
+        return jetInstance.getJobsInt(onlyName, onlyJobId);
     }
 
     private void addToSubmittedJobs(@Nonnull Job job) {
-        getThreadLocalParameters().addSubmittedJob(job);
+        ExecuteJobParameters executeJobParameters = getExecuteJobParameters();
+        executeJobParameters.addSubmittedJob(job);
     }
 
     private void updateJobConfig(JobConfig jobConfig) {
-        ExecuteJobParameters jobParameters = getThreadLocalParameters();
+        ExecuteJobParameters jobParameters = getExecuteJobParameters();
 
         if (jobParameters.hasJarPath()) {
```

Comment:
This throws NPE when `Hazelcast.bootstrapInstance()` is used directly (without submit). See e.g. `com.hazelcast.instance.impl.HazelcastBootstrapTest`

Reply:
rev:. 9bb91a478711f3df8a84329f76425824396a821f